### ISSUE TITLE
1.10.x OSX config file change to follow unix conventions - ~/.config

### ISSUE THREAD

**Issue by wakatara** [gopasspw/gopass#1547]
### Summary

On upgrading to 1.10, the config file for OSX has been changed to reside in `~/Library/Application Support/gopass/`. While this is normal for graphical applications, for command line applications the majority of OSX users expect config files to be in the ~/.config directory.

### Steps To Reproduce

Upgrade to 1.10. Re-run gopass. File moves config file and stores there. Also good for principle of least surprise. 

### Expected behavior

Would expect Gopass to honour a config.yml entry in ~/.config/gopass/config.yml as per linux conventions (also, makes it easier to keep cross platform systems in sync. It should look there before Application Support or offer to place config there vs Application Support on upgrade/"see you're running GoPass for first time"

### Environment

OS: OSX 10.15.6
gopass: 1.10
installation on Mac: homebrew
installation on linux: go get -u ... 
(POP OS 20.04 on linux)

**Comment by dominikschulz** [comment]
Yeah, I don't know. We're trying to follow OS conventions here.

Is there any authoritative guidance on where CLI config files should be placed on MacOS?

**Comment by wakatara** [comment]
Not that I am aware of for CLI apps. But, most CLI applications, say installed via homebrew (the de facto package manager on OSX), tend to have either dotfile directory in `$HOME` or more often, a directory in `.config`.

Also, I think it'd make your lives a lot easier if you do what you would do on a normal-ish unix system with osx for CLIs.

I had to actually go hunting for where you had dropped the new files since it broke my 1.9.x install when I upgraded.  =]  (I had my .password-store directory set in $HOME.

**Comment by dominikschulz** [comment]
I really don't have a strong opinion on this. Not sure how to move on with this.

**Comment by wakatara** [comment]
Hmmm... lemme see if I can convince you of the use case. 

The people who will care the most about this are people who are primarily using GoPass as a cross-platform password solution where they might make changes on linux, osx, or iOS.

So, for example, I use this on both my OSX and linux box and want to, as much as possible, keep things like configuration files in sync (for example, like `ln -s` ing to a cloud directory of some sort they have access to, either on Dropbox, GDrive or what have you.). Not doing this means most people like myself will symlink in the Apple directory *back* to a `~/.config/gopass` directory as on a linux system to keep that sane and universal.

(For example, this is the setup I use: https://daryl.wakatara.com/password-management-with-gopass-password-store-gopass-bridge-and-pass-for-ios )

Furthermore, while I can understand how you might want to break from `pass`-es original origins, this is a pretty surprising change. I literally had to hunt around for where you'd buried the new config and store, so in keeping with the principle of minimal surprise... =] 

I'd also love to see a post or such on the new `store/root` structure you guys are using, when I migrated from `passpie` I stuck to `pass`'s original structure so my entire directory structure is in `~/.password-store` (git stored in the clear since gpg... =] ). as I imagine most people's are. 

Suffice it to say, I am surprised more OSX users have not chimed in on this one. I installed via `go get -u github/gopass` but most people using this with homebrew will be a bit surprised when the config file moves to what they normally consider GUI territory. I'm honestly struggling to think of another CLI app that stores config details there. 

(also, leaving it in ~/.config/gopass makes things simpler for you guys from a coding perspective I imagine since all unix based systems can sit that way.).
